### DISCUSSION TITLE
Block grouping in Blocklist

### DISCUSSION THREAD

**Discussion by madsoulswe** [umbraco/Umbraco-CMS#12302] [Features and ideas]
Add a field to enable grouping of blocklists, similar to Propertyeditors.

Here is a simple example
![image](https://user-images.githubusercontent.com/7071187/165236677-eb733d44-1089-46f5-be73-fffbe6afb3dd.png)
![image](https://user-images.githubusercontent.com/7071187/165240441-6b808631-5ffa-4ebc-ab41-256638c14090.png)

**Comment by bjarnef** [comment]
Unlike groups in property editors, we may also need to consider localization here. For example using dictionary items and the syntax "#Header". Honestly I haven't checked if the Label property supports this feature like Label and Description on content types.

**Reply by markadrake** [reply]
I don't know if this is a Umbraco Forms only feature, but "[magic strings](https://our.umbraco.com/documentation/add-ons/umbracoforms/developer/Magic-Strings/)" would be a plausible solution? I would like to see it slowly integrated across all of Umbraco if it could be.

I feel the need to state that the current implementation of "magic strings" seems funky to me from a FED perspective. I'd like to see improvements here too.

**Comment by markadrake** [comment]
Our entire UI library is composed of blocks, and as such, this list can grow unwieldy. Any form of added categorization, sorting, and filtering would be a welcome addition and improve the editor's experience.
